rosplan_planning_system/src/ProblemGeneration/PDDLProblemGenerator.cpp

```diff
@@ -254,6 +252,28 @@ namespace KCL_rosplan {
 	/* goal */
 	/*------*/
 
+    void PDDLProblemGenerator::parseInequalityExpression(rosplan_knowledge_msgs::ExprComposite exprComposite, int size, std::ofstream &pFile){
+        for(size_t i = 0; i < size; i++) {
+
+            switch(exprComposite.tokens[i].expr_type){
+                case 0: pFile << " "<< exprComposite.tokens[i].constant << " "; break;
+                case 1:
+                {
+                    pFile << "(";
+                    pFile << exprComposite.tokens[i].function.name;
+                    for(size_t j = 0; i < exprComposite.tokens[j].function.typed_parameters.size(); j++) {
+                        pFile << exprComposite.tokens[j].function.typed_parameters[j].key+" "+exprComposite.tokens[j].function.typed_parameters[j].value;
+                    }
+                    pFile << ")";
+                } break;
+
+                case 2: pFile << " " << exprComposite.tokens[i].op << " "; break;
+                case 3: pFile << pFile << "(";exprComposite.tokens[i].special_type; pFile << ")";break;
+
+            }
+        }
+    }
+
```

Comment:
There is already a function to print expressions below (printExpression).
Note that the op needs to be converted from the short int to a string, e.g., 0 -> "(<="
Also that closing braces for operators need printed, to get output such as:
"(* (- (+ w x) y) (- 0 z))"

Reply:
Deleted the duplicate function and used the previous one